### PR TITLE
Adds server-side request id when not present and improves error message

### PR DESCRIPTION
**Pull request by osklyar**

### CODE REVIEW

File: rpc/src/main/java/io/teris/rpc/ServiceDispatcherImpl.java [hunk 1]
```diff
@@ -46,6 +50,8 @@
 
 		private ExecutorService executors = null;
 
+		private Supplier<String> uidGenerator = () -> UUID.randomUUID().toString();;
```
Comment (joao-rebelo) [inline rpc/src/main/java/io/teris/rpc/ServiceDispatcherImpl.java:53]: 2 ;;

File: rpc/src/main/java/io/teris/rpc/ServiceDispatcherImpl.java [hunk 2]
```diff
@@ -34,6 +36,8 @@
 
 class ServiceDispatcherImpl implements ServiceDispatcher {
 
+	private final Supplier<String> uidGenerator;
```
Comment (joao-rebelo) [inline rpc/src/main/java/io/teris/rpc/ServiceDispatcherImpl.java:39]: should we call this uuid... for clarity?

**Review by joao-rebelo — APPROVED** [review]
small changes

File: rpc/src/main/java/io/teris/rpc/ServiceDispatcherImpl.java (hunk 1 above)
Comment (osklyar) [reply rpc/src/main/java/io/teris/rpc/ServiceDispatcherImpl.java:53]: uuid is an implementation detail, this can be any unique id

**Pull request merged**